lib/travis/build/script/csharp.rb

```diff
@@ -105,12 +112,29 @@ def mono_repos
           repos
         end
 
+        def mono_osx_url
+          base_url = 'http://download.mono-project.com/archive/'
+
+          case config[:mono]
+          when 'latest'
+            return base_url + 'mdk-latest.pkg'
+          when 'alpha'
+            return base_url + 'mdk-latest-alpha.pkg'
+          when 'beta'
+            return base_url + 'mdk-latest-beta.pkg'
+          when 'weekly', 'nightly'
+            return base_url + 'mdk-latest-weekly.pkg'
+          else
+            return base_url + config[:mono] + "/macos-10-x86/MonoFramework-MDK-#{config[:mono]}.macos10.xamarin.x86.pkg"
+          end
+        end
+
         def mono_version_valid?
           return true if ['latest', 'alpha', 'beta', 'nightly'].include? config[:mono]
           return false unless MONO_VERSION_REGEXP === config[:mono]
 
-          return false if MONO_VERSION_REGEXP.match(config[:mono])[1] == '2' && !is_mono_2_10_8
-          return false if MONO_VERSION_REGEXP.match(config[:mono])[1].to_i < 2
+          return false if MONO_VERSION_REGEXP.match(config[:mono])[1] == '2' && !is_mono_2_10_8 && config[:os] == 'linux'
+          return false if MONO_VERSION_REGEXP.match(config[:mono])[1].to_i < 2 && config[:os] == 'linux'
```

Comment:
@Joshua-Anderson I just tried this and it doesn't work since the folder structure is different. E.g. even for 2.10.1 it's http://download.mono-project.com/archive/2.10.1/macos-10-x86/3/ . I think we should match the Linux behavior of only allowing 2.10.8 for Mac as well once that package is on the download server.